### ISSUE TITLE
CA1822 False positive/conflict with new Extension syntax

### ISSUE THREAD

**Issue by AlgorithmsAreCool** [dotnet/roslyn#78858]
**Version Used**: 
dotnet 10.0.100-preview.4.25258.110
VS2022 Version 17.14.4 Preview 1.0

**Steps to Reproduce**:

```csharp
public static class MyExtensions
{
    extension (string s)
    {
        public int CountProp => s.Length; //Member 'CountProp' does not access instance data and can be marked as static
        public int CountMeth() => s.Length; //Member 'CountMeth' does not access instance data and can be marked as static
    }
}
```

**Diagnostic Id**:

CA1822

**Expected Behavior**:

No CA1822 since this property/method is accessing the receiver as if it was an instance method or whatever the right word is in compiler speak

**Actual Behavior**:
CA1822

**Comment by znakeeye** [comment]
If you disable this warning, you instead get [CA1034: Nested types should not be visible](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1034). Verified with Preview 6.

So now I have to disable two warnings...

**Comment by RikkiGibson** [comment]
https://github.com/dotnet/roslyn-analyzers/blob/8999f33adf45de679fa025534a5e4065e2d30fb1/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.cs#L123

Perhaps this block, which handles primary constructor parameters, should also handle extension parameters.
